static/js/recent_topics.js

```diff
@@ -168,6 +170,32 @@ exports.update_topic_unread_count = function (message) {
     exports.inplace_rerender(topic_key);
 }
 
+exports.toggle_star_topic = function (stream_id, topic) {
+    const key = stream_id + ":" + topic;
+    const topic_row = get_topic_row(key);
+    const topic_data = topics.get(key);
+    const star_icon = topic_row.find(".on_hover_topic_star");
+    if (star_icon.hasClass('fa-star')) {
+        starred_messages.unstar_topic(stream_id, topic);
+    } else {
+        const message = message_store.get(topic_data.last_msg_id);
+        message_flags.toggle_starred_and_update_server(message);
+    }
+};
+
+exports.update_topic_starred_status = function (message_id) {
+    // This function is made to be topic icon starred state independent.
+    const message = message_store.get(message_id);
+    const topic_key = message.stream_id + ":" + message.topic;
+    const topic_row = get_topic_row(topic_key);
+    const star_icon = topic_row.find(".on_hover_topic_star");
+    if (starred_messages.is_topic_starred(message.stream_id, message.topic)) {
+        star_icon.attr('class', 'star fa on_hover_topic_star fa-star');
+    } else {
+        star_icon.attr('class', 'star fa on_hover_topic_star fa-star-o empty-star');
+    }
```

Comment:
Why not write this with `addClass/removeClass` and avoid duplicating the list of 4 classes that are fixed?

Reply:
I updated it to better handle fixed classes. I want to keep it this was as it is easy to read and is robust.